### PR TITLE
Ignoring pre tag immediately following ID 'idl-index'

### PR DESCRIPTION
**Pull request by m-cheung**
W3C Web Specs often contain many smaller IDL Fragments usually containing one or two interfaces throughout the page. At the bottom of each spec page, there is often an `IDL Index` paragraph which provides a summary of all of the fragments present within the page. Unfortunately, this means that when it gets parsed, we get multiple declarations of the same interface. Additionally, the `pre` tags do not differentiate between index or fragment. 

In order to get around this issue, we will ignore the `pre` tag that follow any tag with the ID `idl-index`.

### CODE REVIEW

File: test/node/parsing/UIEvent/5 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+dictionary MouseEventInit : EventModifierInit {
```
Comment (arobins) [inline test/node/parsing/UIEvent/5:1]: The new fragments are unrelated to this change, right?

**Review by arobins — COMMENTED** [review]
Looks a bit gross, but that's normal for a scraper.

File: test/node/parsing/UIEvent/5 (hunk 1 above)
Comment (m-cheung) [reply test/node/parsing/UIEvent/5:1]: Some of the tests use to contain `index` fragments, so I revised just changed the tests to use non-index fragments.

Sidenote: I apologize for the grossness.

File: lib/org/chromium/webidl/IDLFragmentExtractor.js [hunk 2]
```diff
@@ -79,6 +86,17 @@ foam.CLASS({
             return cls && (cls.includes('example') || cls.includes('note'));
           };
 
+          var id = extractAttr(item, 'id');
+
+          // TODO: Replace with more elegant solution.
```
Comment (mdittmer) [inline lib/org/chromium/webidl/IDLFragmentExtractor.js:91]: I think this change is fine, but we should create a doc where we write down over-approximations like this (throughout the pipeline).

File: lib/org/chromium/webidl/IDLFragmentExtractor.js [hunk 3]
```diff
@@ -60,9 +61,15 @@ foam.CLASS({
           // Ignoring all tags. Only extracting text content within pre tags.
           if (isTag) {
             if (item.nodeName === 'pre') {
+              // Finished matching pre tag at this point.
               tagMatching = true;
               tagStack.pop();
-              idlFragments.push(lexer.lib.unescapeString(content));
+
+              if (ignoreNextPre) {
+                ignoreNextPre = false;
```
Comment (mdittmer) [inline lib/org/chromium/webidl/IDLFragmentExtractor.js:69]: Do we need to process other IDL fragments after the index, or does it appear at the end of the document?

Comment (m-cheung) [reply lib/org/chromium/webidl/IDLFragmentExtractor.js:69]: As far as I can tell, the index only appears at the end of the document

**Comment by m-cheung** [comment]
This has been incorporated into #61. Closing this pull request now.

**Pull request closed without merging**